src/detect-http-header.c

```diff
@@ -497,47 +497,11 @@ static void HttpMultiBufHeaderThreadDataFree(void *data)
     SCFree(td);
 }
 
-static InspectionBuffer *GetHttp2HeaderData(DetectEngineThreadCtx *det_ctx,
-        const DetectEngineTransforms *transforms, Flow *f, const uint8_t flags, void *txv,
-        int list_id, uint32_t local_id)
+static bool GetHttp1HeaderData(DetectEngineThreadCtx *det_ctx, const void *txv, const uint8_t flags,
```

Comment:
This is the only callback using `DetectEngineThreadCtx`

Following discussion here https://github.com/OISF/suricata/pull/12972#discussion_r2035861875
is this `DetectThreadCtxGetGlobalKeywordThreadCtx ` the good way to do ?

Reply:
cc @victorjulien